prow/plugins/blunderbuss/blunderbuss.go

```diff
@@ -334,34 +360,74 @@ func findReviewer(ghc githubClient, log *logrus.Entry, useStatusAvailability boo
 			// we've already verified this reviewer is busy
 			continue
 		}
-		busy, err := isUserBusy(ghc, candidate)
-		if err != nil {
-			log.Errorf("error checking user availability: %v", err)
-		}
-		if !busy {
-			return candidate
-		}
-		// if we haven't returned the candidate, then they must be busy.
-		busyReviewers.Insert(candidate)
+		return candidate
 	}
 	return ""
 }
 
-type githubAvailabilityQuery struct {
-	User struct {
-		Login  githubql.String
-		Status struct {
-			IndicatesLimitedAvailability githubql.Boolean
+// prepareLeafReviewersToQuery prepares reviewers list using info from reviewers client crossed with author set
+func prepareLeafReviewersToQuery(rc reviewersClient, ghc githubClient, files []github.PullRequestChange, authorSet sets.String) []string {
+	ownersSeen := sets.NewString()
+	reviewersToQuery := layeredsets.NewString()
+	for _, file := range files {
+		ownersFile := rc.FindReviewersOwnersForFile(file.Filename)
+		if ownersSeen.Has(ownersFile) {
+			continue
 		}
-	} `graphql:"user(login: $user)"`
+		ownersSeen.Insert(ownersFile)
+		reviewers := layeredsets.NewString(rc.LeafReviewers(file.Filename).List()...).Difference(authorSet)
+		reviewersToQuery = reviewersToQuery.Union(reviewers)
+	}
+	return reviewersToQuery.List()
 }
 
-func isUserBusy(ghc githubClient, user string) (bool, error) {
-	var query githubAvailabilityQuery
-	vars := map[string]interface{}{
-		"user": githubql.String(user),
+type githubUserAvailability struct {
+	Login  string
+	Status struct {
+		IndicatesLimitedAvailability bool
+	}
+}
+
+// graphqlTag helps in formating the graphql tag
+func graphqlTag(value string) reflect.StructTag {
+	return reflect.StructTag(strconv.AppendQuote([]byte("graphql:"), value))
+}
+
+// findBusyReviewers finds all the people that are busy and excluded from the review
+func findBusyReviewers(ghc githubClient, log *logrus.Entry, busyReviewers *sets.String, reviewers []string) {
+	const (
+		// will panic if too many fields with too much data in the reflect struct, this setting is very safe
+		githubUserAvailabilityStructLimit = 300
+	)
+	currentIndex := 0
+	numberOfBatches := int(math.Ceil(float64(len(reviewers)) / githubUserAvailabilityStructLimit))
+	for batch := 0; batch < numberOfBatches; batch++ {
+		var fields []reflect.StructField
+		batchError := false
+		structsAdded := 0
+		for i := currentIndex; i < len(reviewers) && structsAdded < githubUserAvailabilityStructLimit; i++ {
+			fields = append(fields, reflect.StructField{
```

Comment:
Instead of predefining a whole query as a struct,` reflect.StructField` is used to compose a dynamic query with the requested number of users. It is parsed by `qhc.query` function on almost the same basis as the predefined struct is parsed.

Reply:
This is the library's author's recommendation. Please refer to https://github.com/shurcooL/githubv4/issues/17.
Also, dynamic query example is pasted below:
```
{
  user1: user(login: "first-user") {
    id
    status {
      indicatesLimitedAvailability
    }
  }
  user2: user(login: "user") {
    id
    status {
      indicatesLimitedAvailability
    }
  }
  user3: user(login: "another-user") {
    id
    status {
      indicatesLimitedAvailability
    }
  }
}
```